src/schemas.jl

```diff
@@ -254,35 +254,79 @@ accepted_field_type(::SchemaVersion, ::Type{Symbol}) = Union{Symbol,String}
 """
     Legolas.find_violation(ts::Tables.Schema, sv::Legolas.SchemaVersion)
 
-For required field `f::F` of `sv`:
+Return first schema violation. For each required field `f::F` of `sv`:
 
 - Define `A = Legolas.accepted_field_type(sv, F)`
 - If `f::T` is present in `ts`, ensure that `T <: A` or else immediately return `f::Symbol => T::DataType`.
 - If `f` isn't present in `ts`, ensure that `Missing <: A` or else immediately return `f::Symbol => missing::Missing`.
 
 Otherwise, return `nothing`.
 
-See also: [`Legolas.validate`](@ref), [`Legolas.complies_with`](@ref)
+See also: [`Legolas.find_violations`](@ref), [`Legolas.validate`](@ref), [`Legolas.complies_with`](@ref)
 """
 find_violation(::Tables.Schema, sv::SchemaVersion) = throw(UnknownSchemaVersionError(sv))
 
 function _find_violation end
 
+"""
+    Legolas.find_violations(ts::Tables.Schema, sv::Legolas.SchemaVersion)
+
+Return vector of all schema violations. For required fields `f::F` of `sv`:
+
+- Define `A = Legolas.accepted_field_type(sv, F)`
+- If `f::T` is present in `ts`, ensure that `T <: A` or else append `f::Symbol => T::DataType` to output vector.
+- If `f` isn't present in `ts`, ensure that `Missing <: A` or else append `f::Symbol => missing::Missing` to output vector.
+
+Otherwise, return `Any[]`.
+
+See also: [`Legolas.find_violation`](@ref), [`Legolas.validate`](@ref), [`Legolas.complies_with`](@ref)
+"""
+find_violations(::Tables.Schema, sv::SchemaVersion) = throw(UnknownSchemaVersionError(sv))
+
+function _find_violations end
```

Comment:
tbh I'm not sure what the role of this is---I added it because I saw there was a `function _find_violation end` above. Thoughts on why the above one exists (and therefore whether this one should)?